fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayBanner.kt

```diff
@@ -7,65 +7,190 @@ package org.mozilla.fenix.tabstray
+import androidx.compose.material.TabRow
 import androidx.compose.material.Text
+import androidx.compose.material.ripple.LocalRippleTheme
+import androidx.compose.material.ripple.RippleAlpha
+import androidx.compose.material.ripple.RippleTheme
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
+import mozilla.components.ui.tabcounter.TabCounter
+import org.mozilla.fenix.R
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
-import org.mozilla.fenix.compose.button.PrimaryButton
 import org.mozilla.fenix.theme.FirefoxTheme
+import java.text.NumberFormat
 
 /**
  * Top-level UI for displaying the banner in [TabsTray].
  *
  * @param isInMultiSelectMode Whether the tab list is in multi-select mode.
+ * @param selectedPage The active [Page] of the Tabs Tray.
+ * @param normalTabCount The total amount of normal browsing tabs currently open.
  * @param onTabPageIndicatorClicked Invoked when the user clicks on a tab page indicator.
  */
 @Composable
 fun TabsTrayBanner(
     isInMultiSelectMode: Boolean,
+    selectedPage: Page,
+    normalTabCount: Int,
     onTabPageIndicatorClicked: (Page) -> Unit,
 ) {
     if (isInMultiSelectMode) {
         MultiSelectBanner()
     } else {
         SingleSelectBanner(
             onTabPageIndicatorClicked = onTabPageIndicatorClicked,
+            selectedPage = selectedPage,
+            normalTabCount = normalTabCount,
         )
     }
 }
 
+@Suppress("LongMethod")
 @Composable
 private fun SingleSelectBanner(
+    selectedPage: Page,
+    normalTabCount: Int,
     onTabPageIndicatorClicked: (Page) -> Unit,
 ) {
-    Row(
-        modifier = Modifier
-            .fillMaxWidth()
-            .height(80.dp)
-            .padding(all = 16.dp),
-        horizontalArrangement = Arrangement.spacedBy(16.dp),
-    ) {
-        Box(modifier = Modifier.weight(1.0f)) {
-            PrimaryButton(text = "Normal tabs") {
-                onTabPageIndicatorClicked(Page.NormalTabs)
-            }
-        }
+    val normalTabCountText: String
+    val normalTabCountTextModifier: Modifier
+    if (normalTabCount > TabCounter.MAX_VISIBLE_TABS) {
+        normalTabCountText = TabCounter.SO_MANY_TABS_OPEN
+        normalTabCountTextModifier = Modifier.padding(bottom = 1.dp)
+    } else {
+        normalTabCountText = NumberFormat.getInstance().format(normalTabCount.toLong())
```

Comment:
Is this necessary, given that we are just using integers? Seems like `"$normalTabCount"` could suffice

Reply:
Probably not. I admittedly reused some of the logic from [`TabCounterButton`](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/compose/tabstray/src/main/java/mozilla/components/compose/tabstray/TabCounterButton.kt), an already-existing Composable that's unused. There may have been a reason for this formatter logic when that component was made in 2021 that is no longer valid today.